In [144]:
import pandas as pd
import numpy as np
import json

In [145]:
results_path = "../results/seat/seat_m-GPT2Model_c-gpt2-m-kelm.json"
results_dict = json.load(open(results_path))

In [146]:
results_df = pd.DataFrame(results_dict)
results_df.head()

experiment_id                                test  \
0  seat_m-GPT2Model_c-gpt2-m-kelm        angry_black_woman_stereotype   
1  seat_m-GPT2Model_c-gpt2-m-kelm      angry_black_woman_stereotype_b   
2  seat_m-GPT2Model_c-gpt2-m-kelm     heilman_double_bind_competent_1   
3  seat_m-GPT2Model_c-gpt2-m-kelm  heilman_double_bind_competent_1+3-   
4  seat_m-GPT2Model_c-gpt2-m-kelm    heilman_double_bind_competent_1-   

    p_value  effect_size  
0  0.379840     0.117114  
1  0.100000     0.844286  
2  0.765190    -0.369982  
3  0.896814    -0.643211  
4  0.967599    -0.918774

In [147]:
GENDER_TESTS = [
    "sent-weat6",
    "sent-weat6b",
    "sent-weat7",
    "sent-weat7b",
    "sent-weat8",
    "sent-weat8b",
]

RACE_TESTS = [
    "sent-angry_black_woman_stereotype",
    "sent-angry_black_woman_stereotype_b",
    "sent-weat3",
    "sent-weat3b",
    "sent-weat4",
    "sent-weat5",
    "sent-weat5b",
]

RELIGION_TESTS = [
    "sent-religion1",
    "sent-religion1b",
    "sent-religion2",
    "sent-religion2b",
]


In [148]:
# Average absolute SEAT effect size
avg_eff_size_dict = {}
for test_list, category in zip([GENDER_TESTS, RELIGION_TESTS, RACE_TESTS], ["gender", "religion",
                                                                      "race"]):
    category_df = results_df[results_df["test"].isin(test_list)]
    avg_eff_size_dict[category] = sum(category_df["effect_size"]\
        .map(abs)) / len(category_df)

avg_eff_size_dict

{'gender': 0.39659028731040724,
 'religion': 0.23385283270820106,
 'race': 0.2958328234756537}

In [149]:
sum(avg_eff_size_dict.values()) / 3

0.30875864783142065

Results:

BERT base: 0.577395465618045
GPT2 medium: 0.25635265762214704
GPT2 medium KELM: 0.30875864783142065
Roberta base: 0.45789884436915695
Luke base: 0.4055710669891146
Colake base: 0.33373944890951185